hypothesis/client
# Remove `OAUTH_TOKENS_CHANGED` event

~~**Depends on https://github.com/hypothesis/client/pull/2125**~~

Remove the last remaining use of the `$rootScope` service to broadcast /
subscribe to the `OAUTH_TOKENS_CHANGED` event. This event is emitted when the user
has the client active in multiple tabs and they change their login in one tab, requiring other tabs
to update to reflect the change.

Instead the `auth` service is now an `EventEmitter` and the `session`
service subscribes directly to `oauthTokensChanged` events from it.
Whereas we replaced most other events with observers which react to data
changing in the store, this particular use case doesn't lend itself to
that and it makes more sense for the `auth` service to emit events to
notify interested consumers.

Fixes https://github.com/hypothesis/client/issues/1994.

----

**Testing this from a user's POV:**

1. Open two tabs at a page with the Hypothesis client embedded.
2. Log in (or log out) in one tab and check that the change is reflected in the other
3. Repeat step 2 but from the other tab

Note that if the two tabs have separate top-level domains then some browsers will isolate the client's local storage so this will not work. That's a known limitation that is not changed in this PR.
